addons/sale_loyalty/tests/test_loyalty.py

```diff
@@ -310,6 +328,262 @@ def test_discount_max_amount_on_specific_product(self):
         msg = "Reward amount should be the percentage one if under the max amount discount."
         self.assertEqual(reward_amount_tax_included, -6, msg)
 
+    def test_points_awarded_global_discount_code_no_domain_program(self):
+        """
+        Check the calculation for points awarded when there is a global discount applied and the
+        loyalty program applies on all products (no domain).
+        """
+        loyalty_program = self.env['loyalty.program'].create(
+            self.env['loyalty.program']._get_template_values()['loyalty']
+        )
```

Comment:
Nit: is you use an additional variable, you could fit this on a single line :)
```suggestion
        LoyaltyProgram = self.env['loyalty.program']
        loyalty_program = LoyaltyProgram.create(LoyaltyProgram._get_template_values()['loyalty'])
```